packages/web-components/src/button/button.ts

```diff
@@ -0,0 +1,103 @@
+import { attr } from '@microsoft/fast-element';
+import { FASTButton } from '@microsoft/fast-foundation';
+import { ButtonAppearance, ButtonShape, ButtonSize } from './button.options.js';
+
+/**
+ * The base class used for constructing a fluent-button custom element
+ * @public
+ */
+export class Button extends FASTButton {
+  /**
+   * The appearance the button should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: appearance
+   */
+  @attr
+  public appearance?: ButtonAppearance | undefined;
+
+  /**
+   * The shape the button should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: shape
+   */
+  @attr
+  public shape?: ButtonShape | undefined;
+
+  /**
+   * The size the button should have.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: size
+   */
+  @attr
+  public size?: ButtonSize;
+
+  /**
+   * The button has an icon
+   * This attribute is not required if icon-only is true
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: icon
+   */
+  @attr({ mode: 'boolean' })
+  public icon: boolean = false;
```

Comment:
@miroslavstastny I agree here we'd prefer to not have this. The issue here isn't actually the padding between content, that's applied without this. The padding change here is *vertical* because of the size of the icon, which makes it problematic. Long term goal here is to leverage `:host(:has())` to address this for now. An alternative here is that we can try to observe children and use JS magic in the case that those selectors aren't supported.

I can try and adjust the CSS, but the issue is ultimately that when there *is not* an icon present, there is 3px of padding and when there *is* an icon (icon-only, start, or end) the padding becomes 1px. :(

Reply:
Here's a super simple way to illustrate the problem (where I'm pointing to the ideal CSS solution): https://github.com/w3c/webcomponents-cg/issues/5#issuecomment-1469174859

Reply:
Let me see if I can do some magic to handle this without forking the FAST template...

Reply:
min-height!